synapse/rest/media/v1/preview_url_resource.py

```diff
@@ -676,7 +676,11 @@ async def _expire_url_cache_data(self):
             logger.debug("No media removed from url cache")
 
 
-def decode_and_calc_og(body, media_uri, request_encoding=None):
+def decode_and_calc_og(body, media_uri, request_encoding=None) -> Dict[str, str]:
+    # If there's no body, nothing useful is going to be found.
+    if not body:
+        return {}
```

Comment:
An alternative might be to check if `tree` (below) is `None` before passing it into `_calc_og`.